src/psyclone/core/signature.py

```diff
@@ -105,11 +105,34 @@ def __eq__(self, other):
             return False
         return self._signature == other._signature
 
+    # ------------------------------------------------------------------------
+    def __ne__(self, other):
+        '''Required for != comparisons of Signatures with python2.
+        Compares two objects (one of which might not be a Signature).'''
```

Comment:
@arporter I've been thinking about this, and cant' make up my mind (except that I agree it needs to be improved).
Instead of raising a type error, we would just handle the case that the other instance is not a Signature - just return the result of a comparison between `self` and `other` directly, or `self._signature` and `other`.

I admit I just don't like that a comparison should ever raise an exception (and obviously my code did ;) ), but I also can't think of a single good reason why someone should ever compare a signature with ... anything else really.

So I just thought I write this, and re-confirm with you that you prefer to have a TypeError raised when a signature is compared with a non-signature :)

Reply:
Thanks Joerg. I've just fired up python and tried:

    >>> 3 == "hello"
    False
    >>> 3 < "hello"
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
    TypeError: '<' not supported between instances of 'int' and 'str'

so I think your approach for `==` and `!=` is probably right but we should raise a TypeError for the other comparison operations.

Reply:
Why didn't I think of doing that?? That must be while they are paying you the big bucks :)

Fixed, == and != won't raise an error now, everything else does.

Reply:
Are they paying me big bucks? I'll have to check :-)